src/test/java/net/snowflake/client/jdbc/telemetryOOB/TelemetryServiceIT.java

```diff
@@ -231,4 +240,76 @@ public void stressTestCreateUrgentEvent()
     }
     sw.stop();
   }
+
+  private void generateDummyException(int value, SFSession session) throws SnowflakeSQLLoggedException
+  {
+    if (value > 0)
+    {
+      String queryID = "01234567-1234-1234-1234-00001abcdefg";
+      String reason = "This is a test exception.";
+      String sqlState = SqlState.NO_DATA;
+      int vendorCode = value;
+      throw new SnowflakeSQLLoggedException(queryID, reason, sqlState, vendorCode, session);
+    }
+  }
+  
+  /**
+   * Test case for checking telemetry message for SnowflakeSQLExceptions. Assert that telemetry OOB endpoint is reached
+   * after a SnowflakeSQLLoggedException is thrown.
+   * @throws SQLException
+   */
+  @Test
+  public void testSnowflakeSQLLoggedExceptionOOBTelemetry() throws SQLException, InterruptedException
+  {
+    // make a connection to initialize telemetry instance
+    Connection con = getConnection();
+    int fakeErrorCode = 27;
+    try
+    {
+      generateDummyException(fakeErrorCode, null);
+    }
+    catch (SnowflakeSQLLoggedException e)
+    {
+      // The error response has the same code as the the fakeErrorCode
+      assertThat("Communication error", e.getErrorCode(),
+              equalTo(fakeErrorCode));
+
+      // since it returns normal response,
+      // the telemetry does not create new event
+      Thread.sleep(WAIT_FOR_TELEMETRY_REPORT_IN_MILLISECS);
```

Comment:
5 seconds seems too long for some case. You can have a while loop here to check the value until it matches or reach the max waiting time.

Reply:
The code after this is checking to see if a telemetry error has been reported. The default value of number of telemetry errors is 0, and the assert is checking to make sure the number of errors stays at 0. So if we don't wait at all, an error might go undetected. This is how other Junit tests such as **testHttpsLoginTimeoutWithOutSSL()** are designed as well. However, I've reduced the wait time from 5 to 2 seconds.

Reply:
Can you run this times for 100 times and make sure it always fit into 2 seconds? We don't want this become a flaky test. Also, why not check the success count? If you check thee success count, you can have a short loop and once it increases you can stop the test.

Reply:
I think the original test worked and was not flaky, since it was based off of all the other telemetry tests we have (testHttpsLoginTimeoutWithSSL(), testHttpsLoginTimeoutWithOutSSL(), testWrongHostNameTimeout(), etc).  Testing for the success count is what led to all those flaky test failures on JDBC so we had to switch to checking the failure count instead. I think the best option so I can get this PR merged in time is to leave the test how it was and then improve all telemetry tests at once with a new Jira task.